src/Compilers/CSharp/Portable/Lowering/AsyncRewriter/AsyncRewriter.cs

```diff
@@ -92,29 +92,29 @@ internal partial class AsyncRewriter : StateMachineRewriter
         /// </returns>
         protected bool VerifyPresenceOfRequiredAPIs()
         {
-            DiagnosticBag bag = DiagnosticBag.GetInstance();
+            var bag = BindingDiagnosticBag.GetInstance();
 
             VerifyPresenceOfRequiredAPIs(bag);
 
             bool hasErrors = bag.HasAnyErrors();
-            if (hasErrors)
+            if (!hasErrors)
             {
-                diagnostics.AddRange(bag);
+                bag.DiagnosticBag.Clear();
```

Comment:
Perhaps add a clarifying comment that we still want to record assembly usage in the presence of errors.

Reply:
> Perhaps add a clarifying comment that we still want to record assembly usage in the presence of errors.

Moving to using AddDependencies API instead 

---
In reply to: [362137536](https://github.com/dotnet/roslyn/pull/40560#discussion_r362137536) [](ancestors = 362137536)

Reply:
This is addressed in [https://github.com/dotnet/roslyn/pull/40640](https://github.com/dotnet/roslyn/pull/40640)

---
In reply to: [362138345](https://github.com/dotnet/roslyn/pull/40560#discussion_r362138345) [](ancestors = 362138345,362137536)